optaplanner-docs/src/main/asciidoc/ConstraintStreams/ConstraintStreams-chapter.adoc

```diff
@@ -176,6 +176,31 @@ The following constraint stream cardinalities are currently supported:
 |4          |  Quad|`QuadConstraintStream<A, B, C, D>`
 |===
 
+[[constraintStreamsHigherCardinalities]]
+=== Achieving higher cardinalities
+
+OptaPlanner currently does not support constraint stream cardinalities higher than 4.
+However, with <<constraintStreamsMappingTuples,tuple mapping>> it is easy to achieve effectively infinite cardinality:
+
+[source,java,options="nowrap"]
+----
+    private Constraint pentaStreamExample(ConstraintFactory factory) {
+        return factory.from(Shift.class) // UniConstraintStream<Shift>
+                .join(Shift.class)       // BiConstraintStream<Shift, Shift>
+                .join(Shift.class)       // TriConstraintStream<Shift, Shift, Shift>
+                .join(Shift.class)       // QuadConstraintStream<Shift, Shift, Shift, Shift>
+                .map(Tuple::of)          // UniConstraintStream<Tuple<Shift, Shift, Shift, Shift>>
+                .join(Shift.class)       // BiConstraintStream<Tuple<Shift, Shift, Shift, Shift>, Shift>
+                ...                      // This BiConstraintStream carries 5 Shift elements.
+    }
+----
+
+[NOTE]
+====
+OptaPlanner does not provide any `Tuple` implementations out of the box.
+We recommend using one of the myriad freely available 3rd party implementations.
```

Comment:
Avoid "we" in docs. Avoid the tense of "using" in favor of active:

```suggestion
It's recommended to use one of the myriad freely available 3rd party implementations.
```

Ps: What does "myriad" mean? Many docs readers aren't native English speakers. It's ok to use difficult words if they describe an property of the behavior (such as "bijective"), as it is explained immediately. But in normal sentences, like this, maybe we should avoid uncommon English words without explanation unless they bring enough value. Wdyt?